pkg/machines/scripts/install_machine.sh

```diff
@@ -89,10 +89,20 @@ fi
 # backup
 DOMAIN_FILE="`mktemp`"
 
-virsh -c "$CONNECTION_URI" -q destroy "$VM_NAME" 2>/dev/null
+virsh -c "$CONNECTION_URI" -q destroy "$VM_NAME" 2>/dev/null || true
 virsh -c "$CONNECTION_URI" -q dumpxml "$VM_NAME" > "$DOMAIN_FILE"
 virsh -c "$CONNECTION_URI" -q undefine "$VM_NAME" --managed-save
 
+handleFailure() {
+    # return back if failed
+    set +e
+    if vmExists "$VM_NAME"; then
+        virsh -c "$CONNECTION_URI" -q define "$DOMAIN_FILE"
+    fi
```

Comment:
That's the bit that I still don't understand -- Isn't this supposed to *clean up* the broken VM? Also, if it already exists, why does it need to get re-defined again? This should at least get a comment.

Reply:
If the virt-install fails, and there was an seperate install step, we need to redefine the VM in order to get the metadata back that show the install button again.

Anyway - I will not merge this one yet - the tests are virtually green - some machines tests were just skipped on known issues.